src/Microsoft.ML.Data/Transforms/HashTransform.cs

```diff
@@ -670,28 +670,28 @@ private ValueGetter<VBuffer<uint>> ComposeGetterVec(IRow input, int iinfo)
 
         #region Core Hash functions, with and without index
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private static uint HashCore(uint seed, ref DvText value, uint mask)
+        private static uint HashCore(uint seed, ref ReadOnlyMemory<char> value, uint mask)
         {
             Contracts.Assert(Utils.IsPowerOfTwo(mask + 1));
-            if (!value.HasChars)
+            if (value.IsEmpty)
                 return 0;
-            return (value.Trim().Hash(seed) & mask) + 1;
+            return (ReadOnlyMemoryUtils.Hash(value.Span.Trim(), seed) & mask) + 1;
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private static uint HashCore(uint seed, ref DvText value, int i, uint mask)
+        private static uint HashCore(uint seed, ref ReadOnlyMemory<char> value, int i, uint mask)
         {
             Contracts.Assert(Utils.IsPowerOfTwo(mask + 1));
-            if (!value.HasChars)
+            if (value.IsEmpty)
                 return 0;
-            return (value.Trim().Hash(Hashing.MurmurRound(seed, (uint)i)) & mask) + 1;
```

Comment:
Same as above here - used to call `DvText.Trim()` but now are calling `Span.Trim()`, which changes behavior and perf.